src/openrct2-ui/windows/Finances.cpp

```diff
@@ -379,19 +379,19 @@ class FinancesWindow final : public Window
         }
 
         // Expenditure / Income values for each month
-        int16_t currentMonthYear = static_cast<int16_t>(gDateMonthsElapsed);
+        int32_t currentMonthYear = gDateMonthsElapsed;
         for (int32_t i = SummaryMaxAvailableMonth(); i >= 0; i--)
         {
             screenCoords.y = 0;
 
-            int16_t monthyear = currentMonthYear - i;
+            int32_t monthyear = currentMonthYear - i;
             if (monthyear < 0)
                 continue;
 
             // Month heading
             auto ft = Formatter();
             ft.Add<StringId>(STR_FINANCES_SUMMARY_MONTH_HEADING);
-            ft.Add<uint16_t>(monthyear);
+            ft.Add<int32_t>(monthyear);
```

Comment:
I am not sure you can do this without editing the string first to take a 32-bit value.

Reply:
`STR_FINANCES_SUMMARY_MONTH_HEADING` formats `{MONTH}` and it seems like formatter takes int32_t for the argument?

```c++
case FormatToken::Month:
    if constexpr (std::is_integral<T>())
    {
        auto szMonth = language_get_string(DateGameMonthNames[date_get_month(arg)]);
        ...
    }
```

```c++
int32_t date_get_month(int32_t months)
{
    return months % MONTH_COUNT;
}
```

Reply:
I think, unless you call the template version of FormatString, this will be uint16 still. See:
https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/localisation/Formatting.cpp#L778

I could be wrong though, have you tried running this code with larger than uint16 month numbers?

Reply:
Yes I tried with month bigger than 0x10000 and it works find.
Maybe `BuildAnyArgListFromLegacyArgBuffer` is not called in this case?

Reply:
Oh no, I was wrong.
`BuildAnyArgListFromLegacyArgBuffer` is called and month value is read as uint16_t.
However final month value is `months % MONTH_COUNT`, so it was the same by accident.

I will modify `BuildAnyArgListFromLegacyArgBuffer` as well.

Reply:
Actually, it looks like keeping {MONTH} as uint16 is simpler than making it 32bit wide...!